### PR TITLE
Use revive instead of golint

### PR DESCRIPTION
**Pull request by pigletfly**
Signed-off-by: pigletfly <wangbing.adam@gmail.com>

# Description

The linter 'golint' is deprecated (since v1.41.0).ref: https://github.com/golangci/golangci-lint/issues/1892

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
We need to keep the same consistent code style across all projects, so please open an issue for this and track it.

**Comment by pigletfly** [comment]
> We need to keep the same consistent code style across all projects, so please open an issue for this and track it.
> 
> Honestly, I do not want to make this change, since no useful and necessary linter was introduced.

Task list is created, https://github.com/dapr/components-contrib/issues/1687.As said in the description, `revive`  is a drop-in replacement, `golint` repo is archived already.

File: .golangci.yml [hunk 1]
```diff
@@ -88,7 +88,7 @@ linters-settings:
     disable:
       - shadow
     disable-all: false
-  golint:
+  revive:
     # minimal confidence for issues, default is 0.8
     min-confidence: 0.8
```
Comment (shubham1172) [inline .golangci.yml:93]: Should `min-confidence` change to `confidence`? 
https://github.com/mgechev/revive#configuration
```
# Sets the default failure confidence. This means that linting errors
# with less than 0.8 confidence will be ignored.
```

**Comment by shubham1172** [comment]
> We need to keep the same consistent code style across all projects, so please open an issue for this and track it.
> 
> Honestly, I do not want to make this change, since no useful and necessary linter was introduced.

golint is now deprecated in favor of revive. We should update it here too.

File: bindings/http/http.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ import (
 )
 
 // HTTPSource is a binding for an http url endpoint invocation
-// nolint:golint
+//revive:disable
```
Comment (shubham1172) [inline bindings/http/http.go:35]: Comment valid for all changes in this PR.

`//nolint:golint` used to either work for one specific block, or for a single line if specified inline.

`//revive:disable` comments are valid for the entire file. They need to be enabled again with `//revive:enable` after the block ends. For inline usage, we can use `//revive:disable-next-line`.

File: state/redis/redis.go [hunk 3]
```diff
@@ -403,7 +403,7 @@ func (r *StateStore) Multi(request *state.TransactionalStateRequest) error {
 
 	pipe := r.client.TxPipeline()
 	for _, o := range request.Operations {
-		//nolint:golint,nestif
+		//revive:disable,nestif
```
Comment (shubham1172) [inline state/redis/redis.go:406]: Should it be two different comments? `nestif` and `revive` are separate linters.

**Review by berndverst — CHANGES_REQUESTED** [review]
Please address the feedback from @shubham1172

File: bindings/http/http.go (hunk 2 above)
Comment (pigletfly) [reply bindings/http/http.go]: gotcha

**Comment by pigletfly** [comment]
a lot of new lint errors are coming.

**Comment by shubham1172** [comment]
@pigletfly Are the linter errors introduced due to golintci's version upgrade? If yes, we might need to fix them too.

**Comment by pigletfly** [comment]
> @pigletfly Are the linter errors introduced due to golintci's version upgrade? If yes, we might need to fix them too.

I tested locally.It's the golangci-lint version upgrade.

**Comment by shubham1172** [comment]
If there are some linters enabled out of box in the newer versions, we can choose to keep it (unless we are strongly opinionated about it).

**Comment by pigletfly** [comment]
these are new linters:

```
    - tparallel
    - paralleltest
    - wrapcheck
    - tagliatelle
    - ireturn
    - exhaustivestruct
    - errchkjson
    - contextcheck
    - gomoddirectives
    - godot
    - cyclop
    - varnamelen
    - gosec
    - errorlint
    - forcetypeassert
    - ifshort
    - maintidx
    - nilnil
    - predeclared
    - tenv
    - thelper
    - wastedassign
    - containedctx
    - gosimple
```

**Comment by shubham1172** [comment]
@berndverst do you think we should track a separate issue to re-enable the new linters and fix the linting issues in our code?

**Comment by ItalyPaleAle** [comment]
Note that as I just learnt, when updating to Go 1.18 a bunch of linters get disabled. 

On dapr/dapr, golangci-lint now shows this:

```
WARN [linters context] bodyclose is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] nilerr is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] rowserrcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] sqlclosecheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] staticcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] stylecheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] unused is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
```

This has caused MORE linter failures in the PR that switched to Go 1.18 because lint rules such as "nilerr" don't exist anymore, so `//nolint:nilerr` in itself throws an error.

So I wouldn't stress too much about what linters are enabled or disabled here, as when the PR to upgrade to Go 1.18 is merged, we'll have to revisit that anyways.

**Comment by daixiang0** [comment]
We should track linter changes and re-enable them when they are ready for Go 1.18.

**Comment by ItalyPaleAle** [comment]
> We should track linter changes and re-enable them when they are ready for Go 1.18.

For the linters that are not available in 1.18, they are not explicitly disabled, so once golangci-lint is updated and adds them, they should be re-enabled automatically.

For the linters that were [explicitly disabled in this PR](https://github.com/dapr/components-contrib/pull/1685/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9R249-R273), that's a different story.

**Comment by daixiang0** [comment]
Agree, but we need to update golangci-lint version, so still need to track.

**Review by yaron2 — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**